File: Topcoder/Python/UserName.py
class UserName:
  def newMember(self, existingNames, newName):
    existingNames = dict( (x, True) for x in existingNames)
    i = 1
    name = newName
    while name in existingNames:
      name = ''.join([newName, str(i)])
      i += 1
    return name

File: online_judges/Topcoder/Java/SRM 203 DIV 2/src/UserName.java
import java.util.Collections;
import java.util.HashSet;

public class UserName {

  public String newMember(String[] existingNames, String newName) {
    HashSet<String> names = new HashSet<String>();
    Collections.addAll(names, existingNames);
    String copyNewName = newName;
    int con = 1;
    while (names.contains(copyNewName)) {
      copyNewName = newName + "" + (con++);
    }
    return copyNewName;
  }
}
